pymbolic/mapper/unifier.py

```diff
@@ -78,7 +86,10 @@ def __init__(self,
         self.lmap = lmap
         self.rmap = rmap
 
-    def unify(self, other: UnificationRecord):
+    def unify(self, other: UnificationRecord | None) -> UnificationRecord | None:
+        if other is None:
+            return None
```

Comment:
Is this reasonable? `map_commut_assoc` seems to want to pass `None` to this, so I just added it..

Reply:
Sure, that makes sense to me.